decoder_test.go

```diff
@@ -468,17 +439,10 @@ func TestReadCertificate(t *testing.T) {
 		}
 
 		cert, err := d.readCertificate(tt.version)
-		if !reflect.DeepEqual(err, tt.err) {
-			t.Errorf("invalid error '%v' '%v'", err, tt.err)
-		}
-
-		if !reflect.DeepEqual(cert, tt.cert) {
-			t.Errorf("invalid certificate '%v' '%v'", cert, tt.cert)
-		}
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
+		assert.Truef(t, reflect.DeepEqual(cert, tt.cert), "invalid certificate '%v' '%v'", cert, tt.cert)
 
 		hash := d.h.Sum(nil)
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```

Comment:
Please consider using assert.Equal and put expected and actual arguments in a correct order.